core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -989,35 +1023,74 @@ protected boolean shouldExecuteReplication(Settings settings) {
         return IndexMetaData.isIndexUsingShadowReplicas(settings) == false;
     }
 
+    protected static class ShardResponse extends ActionResponse {
+        long localCheckpoint;
```

Comment:
opted with making the class private and leave these as is they are accessed from the outside. I know it's not needed, but I feel it's clearer.